src/pages/taskQueue/TaskQueueTable/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useMemo, useRef, useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { LGColumnDef, useLeafyGreenTable } from "@leafygreen-ui/table";
+import { Body, Disclaimer } from "@leafygreen-ui/typography";
+import { useTaskQueueAnalytics } from "analytics";
+import { StyledRouterLink, WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import {
+  getVersionRoute,
+  getTaskRoute,
+  getUserPatchesRoute,
+} from "constants/routes";
+import { TaskQueueItem, TaskQueueItemType } from "gql/generated/types";
+import { formatZeroIndexForDisplay } from "utils/numbers";
+import { msToDuration } from "utils/string";
+
+type TaskQueueColumnData = Omit<TaskQueueItem, "revision">;
+interface TaskQueueTableProps {
+  taskQueue: TaskQueueColumnData[];
+  taskId?: string;
+}
+const estimateSize = () => 65;
+
+const TaskQueueTable: React.FC<TaskQueueTableProps> = ({
+  taskId,
+  taskQueue = [],
+}) => {
+  const taskQueueAnalytics = useTaskQueueAnalytics();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [selectedRowIndexes, setSelectedRowIndexes] = useState<number[]>([]);
+  const columns = useMemo(
+    () => taskQueueTableColumns(taskQueueAnalytics.sendEvent),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
+
+  const table = useLeafyGreenTable<TaskQueueColumnData>({
+    data: taskQueue,
+    columns,
```

Comment:
nit. do you know if there's a good way to make the column sizes fixed so they don't keep resizing if you scroll? No worries if not, it's a constant problem I have with the table component in general. 😓